### PR TITLE
(core): convert server group warning service to TS

### PR DESCRIPTION
**Pull request by icfantv**

### CODE REVIEW

**Comment by icfantv** [comment]
@anotherchrisberry PTAL

File: app/scripts/modules/core/serverGroup/details/serverGroupWarningMessage.service.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  static get $inject(): string[] {
+    return ['$templateCache', '$interpolate'];
+  }
+
+  constructor(private $templateCache: ng.ITemplateCacheService,
+              private $interpolate: ng.IInterpolateService) {}
+
+  public getMessage(serverGroup: ServerGroup): string {
+    const template: any = this.$templateCache.get(require('./deleteLastServerGroupWarning.html'));
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/serverGroup/details/serverGroupWarningMessage.service.ts]: Can `template` be cast as a `string` instead of `any`?

Comment (icfantv) [inline app/scripts/modules/core/serverGroup/details/serverGroupWarningMessage.service.ts]: why yes.  yes it can.

**Comment by icfantv** [comment]
rebased and fixed.

**Pull request merged**